src/host/inputBuffer.cpp

```diff
@@ -602,6 +602,27 @@ size_t InputBuffer::Write(const std::span<const INPUT_RECORD>& inEvents)
     }
 }
 
+void InputBuffer::WriteString(const std::wstring_view& text)
+try
+{
+    if (text.empty())
+    {
+        return;
+    }
+
+    const auto initiallyEmptyQueue = _storage.empty();
+
+    _writeString(text);
+
+    if (initiallyEmptyQueue && !_storage.empty())
+    {
+        ServiceLocator::LocateGlobals().hInputEvent.SetEvent();
+    }
+
+    WakeUpReadersWaitingForData();

```

Comment:
`_HandleTerminalInputCallback` doesn't wake up readers if there were no bits - should we do that here?

Reply:
I was a bit afraid of outright using `_HandleTerminalInputCallback`, because `InputBuffer::Write` actually does it exactly like that: It always calls `WakeUpReadersWaitingForData()`. In the future, if I ever get to rewrite `InputBuffer` with proper string support, I was hoping to really understand how this works and to properly unify the event/wakeup code.